src/edu/csus/ecs/pc2/clics/API202306/CLICSScoreboard.java

```diff
@@ -0,0 +1,129 @@
+import edu.csus.ecs.pc2.services.core.JSONUtilities;
+
+/**
+ * CLICS Scoreboard
+ * Contains information about the scoreboard
+ *
+ * @author John Buck
+ *
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CLICSScoreboard {
+
+    @JsonProperty
+    private String time;
+
+    @JsonProperty
+    private String contest_time;
+
+    @JsonProperty
+    private CLICSContestState state;
+
+    @JsonProperty
+    private CLICSScoreboardRow [] rows;
+
+    /**
+     * Fill in the scoreboard information
+     *
+     */
+    public CLICSScoreboard(IInternalContest model, IInternalController controller, Group group, Integer division)  throws IllegalContestState, JAXBException, IOException {
+
+        DefaultScoringAlgorithm scoringAlgorithm = new DefaultScoringAlgorithm();
+
+        Properties properties = ScoreboardUtilities.getScoringProperties(model);
+
+        ArrayList<Group> groupList = null;
+
+        if(group != null) {
+            groupList = new ArrayList<Group>();
+            groupList.add(group);
+        }
+        // legacy - standings are created as XML, and we convert that to JSON.
+        String xml = scoringAlgorithm.getStandings(model, null, division, groupList, properties, StaticLog.getLog());
+
+        ContestStandings contestStandings = ScoreboardUtilities.createContestStandings(xml);
+
+        // This is what we want to return:
+        //        {
+        //            "time": "2014-06-25T14:13:07.832+01",
+        //            "contest_time": "4:13:07.832",
+        //            "state": {
+        //              "started": "2014-06-25T10:00:00+01",
+        //              "ended": null,
+        //              "frozen": "2014-06-25T14:00:00+01",
+        //              "thawed": null,
+        //              "finalized": null,
+        //              "end_of_updates": null
+        //            },
+        //            "rows": [
+        //              {"rank":1,"team_id":"123","score":{"num_solved":3,"total_time":340},"problems":[
+        //                {"problem_id":"1","num_judged":3,"num_pending":1,"solved":false},
+        //                {"problem_id":"2","num_judged":1,"num_pending":0,"solved":true,"time":20},
+        //                {"problem_id":"3","num_judged":2,"num_pending":0,"solved":true,"time":55},
+        //                {"problem_id":"4","num_judged":0,"num_pending":0,"solved":false},
+        //                {"problem_id":"5","num_judged":3,"num_pending":0,"solved":true,"time":205}
+        //              ]}
+        //            ]
+        //          }
+        time = ZonedDateTime.now( ZoneOffset.UTC ).format( DateTimeFormatter.ISO_INSTANT);
+        contest_time = model.getContestTime().getElapsedTimeStr();
+        state = new CLICSContestState(model, null);
+
+        ArrayList<CLICSScoreboardRow>rowsArray = new ArrayList<CLICSScoreboardRow>();
+        HashMap<String, String> probEleToShort = new HashMap<String, String>();
```

Comment:
When I read the name of this variable (and I've seen it in other classes as well), I immediately interpreted it as representing a HashMap which maps _problem elementIds_ (`probEle`) to (java type) `short`.   In fact, it's mapping `ElementId`s to _problem short names_.  For clarity, I would suggest naming the variable (something like) `probEleToShortName`.

Reply:
Since you think it's confusing, I changed it.  Doesn't bother me either way. Changed in next push.